virttest/qemu_devices/qdevices.py

```diff
@@ -907,23 +907,56 @@ def hotplug_qmp(self):
         if params.get("host-nodes"):
             host_nodes = list(map(int, params["host-nodes"].split()))
             params["host-nodes"] = host_nodes
+        for k in params:
+            if params[k] == "yes":
+                params[k] = True
+            if params[k] == "no":
+                params[k] = False
         kwargs = {"qom-type": backend,
                   "id": params.pop("id"),
                   "props": dict(params)}
         return "object-add", kwargs
 
     def verify_unplug(self, out, monitor):
-        out = monitor.info("memory-devices", debug=False)
-        memdev = "/object/%s" % self.get_qid()
-        if memdev not in out:
-            return True
-        return False
+        """
+        :param out: Output of the unplug command
+        :param monitor: Monitor used for unplug
+        :return: True when successful, False when unsuccessful
+        """
+        out = monitor.info("memdev", debug=False)
+        memdev = self.get_qid()
+        for dev in out:
+            if dev["id"] == memdev:
+                return False
+        return True
 
     def verify_hotplug(self, out, monitor):
-        out = monitor.info("memory-devices", debug=False)
-        memdev = "/object/%s" % self.get_qid()
-        if memdev in out:
-            return True
+        """
+        :param out: Output of the hotplug command
+        :param monitor: Monitor used for hotplug
+        :return: True when successful, False when unsuccessful
+        """
+        out = monitor.info("memdev", debug=False)
+        memdev = self.get_qid()
+        params = self.params.copy()
+        for dev in out:
+            if dev["id"] == memdev:
+                if not params.get("host-nodes") and not len(dev["host-nodes"]):
```

Comment:
useless check here

Reply:
line#949  check has cover this check, so  suggest to move  line#949 host node check in here

Reply:
@xutian  You are right that the check for len(dev["host-nodes"]) makes no sense here. I've removed it. But if we move L949 here, we need move L946 too, then the line will exceed 80 chars  even after splitting the if sentence into 2 lines.  I'd like to keep it still.  Thanks.